app/controllers/candidate_interface/sign_in_controller.rb

```diff
@@ -24,12 +24,17 @@ def create
 
     def authenticate
       candidate = FindCandidateByToken.call(raw_token: params[:token])
+      flash[:success] = 'Your apply for teacher training account has been created' if candidate&.last_signed_in_at.nil?
+
       if candidate
         sign_in(candidate, scope: :candidate)
         add_identity_to_log candidate.id
+        candidate.update(last_signed_in_at: Time.zone.now)
```

Comment:
To avoid silent fails:

```suggestion
        candidate.update!(last_signed_in_at: Time.zone.now)
```